context/src/junit/validator.rs

```diff
@@ -197,9 +198,19 @@ pub fn validate(report: &Report) -> JunitReportValidation {
                 ));
             }
 
+            if test_case_validation.level != JunitValidationLevel::Invalid {
+                valid_test_cases.push(test_case.clone());
+            }
+
             test_suite_validation.test_cases.push(test_case_validation);
         }
 
+        if test_suite_validation.level != JunitValidationLevel::Invalid {
+            let mut valid_test_suite = test_suite.clone();
+            valid_test_suite.test_cases = valid_test_cases;
+            report_validation.valid_test_suites.push(valid_test_suite);
+        }
```

Comment:
Do we need to clone these or can we derive them via a method? I think it might be better to do this as a method so as to not take more memory

Reply:
I think that since `validate()` here is working with a reference to a `Report` and not storing any of the test suites / cases nor any identifiers in its iteration, I'm not seeing how we'd be able to derive the valid suites / cases later in a method. Is there a different viable path that you're thinking of?

Reply:
Gotcha. The other way I'd think about it is storing the indexes of these rather than the values themselves. That way we can reference them from the `Report` later on.